app/src/ext_power_generic.c

```diff
@@ -25,8 +28,33 @@ struct ext_power_generic_config {
 struct ext_power_generic_data {
     struct device *gpio;
     bool status;
+#if IS_ENABLED(CONFIG_SETTINGS)
+    bool settings_init;
+#endif
 };
 
+#if IS_ENABLED(CONFIG_SETTINGS)
+static void ext_power_save_state_work(struct k_work *work) {
+    char setting_path[40];
+    struct device *ext_power = device_get_binding(DT_INST_LABEL(0));
```

Comment:
This works. In the future, we'll need to tweak this to put the work inside the driver data, then do something like `strict ext_data data* = CONTAINER_OF(work, struct ext_data, work);` (paraphrasing, I'm on my phone), to get the specific instance this work is for.